pkg/controller/tektoninstallation/tektoninstallation_controller.go

```diff
@@ -0,0 +1,167 @@
+package tektoninstallation
+
+import (
+	"context"
+	toolchainapiv1alpha1 "github.com/codeready-toolchain/api/pkg/apis/toolchain/v1alpha1"
+	"github.com/codeready-toolchain/toolchain-common/pkg/condition"
+	toolchainv1alpha1 "github.com/codeready-toolchain/toolchain-operator/pkg/apis/toolchain/v1alpha1"
+	"github.com/codeready-toolchain/toolchain-operator/pkg/tekton"
+	"github.com/codeready-toolchain/toolchain-operator/pkg/test/toolchain"
+	"github.com/go-logr/logr"
+
+	"github.com/codeready-toolchain/toolchain-operator/pkg/apis/toolchain/v1alpha1"
+	olmv1alpha1 "github.com/operator-framework/operator-lifecycle-manager/pkg/api/apis/operators/v1alpha1"
+	errs "github.com/pkg/errors"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+)
+
+var log = logf.Log.WithName("controller_tektoninstallation")
+
+// Add creates a new TektonInstallation Controller and adds it to the Manager. The Manager will set fields on the Controller
+// and Start it when the Manager is Started.
+func Add(mgr manager.Manager) error {
+	cr := toolchain.NewTektonInstallation()
+	if err := mgr.GetClient().Create(context.TODO(), cr); err != nil && !errors.IsAlreadyExists(err) {
```

Comment:
I thought that this should be in a separated PR - please keep the PRs as small as possible

Reply:
I agree. Let's do the TektonInstallation reconcile logic from CheInstallation controller to TektonInstallation controller in one PR (this one) and creating Installations automaticaly during operator startup in a separate PR.

Reply:
sure, let me make it as a part of separate PR